### ISSUE TITLE
boot-arg ngfxnoaudio does not prevent AppleALC naming of HDAU audio device

### ISSUE THREAD

**Issue by TonyC5** [acidanthera/bugtracker#2178]
Problem: My hackintosh using OC0.8.7 (full specs below) experiences a significantly extended boot cycle when AppleALC renames audio device at pci10de,be3 to HDAU.  I have attempted to use boot-arg "-ngfxnoaudio" documented [here](https://github.com/acidanthera/WhateverGreen/blob/master/Manual/FAQ.GeForce.en.md), but with this boot-arg, AppleALC still names device HDAU.  A search through the AppleALC 1.7.7 and WhateverGreen 1.6.2 source code reveals that the boot-arg is not referenced anywhere in the code.  In fact, a search for both boot-args "-ngfxnoaudio" and "-ngfxnoaudiocon" mentioned in the document and for IOreg properties "no-audio-autofix" and "no-audio-fixconn" reveals that none of these properties are referenced in AppleALC or WhateverGreen source code.  Note that I only want to disable naming of HDAU.  I do not want to disable patching / enabling HDEF.

Testing: Add boot-arg -ngfxnoaudio in OC0.8.7 EFI with AppleALC.kext enabled.  Boot macOS (tested with Big Sur).  Observe that AppleALC has named pci10de,be3 to HDAU.  Disable AppleALC.kext and repeat test.  Without AppleALC.kext, device pci10de,be3 is not named HDAU.

Temporary Fix: In order to achieve the fix that I believe is supposed to be enabled by boot-arg -ngfxnoaudio, I commented out the HDAU naming in AppleALC's kern_alc.cpp:

![1061126667_ScreenShot2022-12-11at10_24_35PM png 39910803bddb7ccf500dcef448667e68](https://user-images.githubusercontent.com/28421182/207413513-8bb2fc79-4609-457e-a721-d515178ff139.png)

When the HDAU naming is commented out in AppleALC, AppleALC does not name pci10de,be3 to HDAU and my hack boots quickly as desired.  With this temporary fix, AppleALC's HDEF functionality is unaffected, so AppleHDA is still able to attach to HDEF and internal audio works as desired.

My system specs:

- Dell Latitude E6410 with Nvidia G3100m graphics
- OpenCore version 0.8.7
- SMBIOS MacModel MacBookPro6,2
- AppleALC version 1.7.7
- My baseline EFI is not using WhateverGreen (I have patched GFX0 with an SSDT), but I tested with and without WhateverGreen 1.6.2.  Regardless of whether I inject WhateverGreen.kext, AppleALC.kext 1.7.7 still renames device HDAU when I use boot-arg -ngfxnoaudio.
- macOS Big Sur 11.7.1

**Comment by TonyC5** [comment]
@abenraj That explains it - thanks.  If this issue is converted to a feature request, I'd like the feature to allow separate enabling/disabling of HDAU rename and HDEF patching (i.e., disabling HDAU rename should not disable HDEF patching and vice versa).

**Comment by vit9696** [comment]
The freeze means your firmware does not initialise HDAU device correctly. I suggest you simply inject class-code FF FF FF FF into your HDAU device and disable it.

**Comment by TonyC5** [comment]
@vit9696 Thank you for the suggestion.  Changing class-code to <FF FF FF FF> does prevent AppleALC renaming of HDAU and thus avoids the slow-boot cycle.

Note that I was not able to overwrite class-code via open core 0.8.7 DeviceProperties.  I needed to change class-code via an SSDT.

This did not work:
![Screen Shot 2023-01-03 at 9 17 02 AM](https://user-images.githubusercontent.com/28421182/210374990-deec34a2-4eaf-45bd-941b-dc7405cc55e8.png)

With open core DeviceProperties, I confirmed that I am able to add new DeviceProperties (those not already defined) to device pci10de,be3, but I am not able to overwrite existing properties (even when I add a Delete entry).

Thank you again for the suggestion.
